api/turing/cluster/spark.go

```diff
@@ -128,6 +129,20 @@ func createSparkRequest(request *CreateSparkRequest) (*apisparkv1beta2.SparkAppl
 	}, nil
 }
 
+func addEnvVars(request *CreateSparkRequest) []apicorev1.EnvVar {
```

Comment:
This is a confusing naming of this function (add env vars to what or where?). Also it's hiding the fact, that default variables are added to user-defined variables here. I'd suggest to refactor it to be explicit.

```golang
func getEnvVarFromRequest(request *CreateSparkRequest) []apicorev1.EnvVar {
	var envVars []apicorev1.EnvVar
	for key, value := range request.EnvironmentVariables {
		envVars = append(envVars, apicorev1.EnvVar{key, value})
	}

	return envVars
}

// And then, combine user-defined vars with default variables where you need them

Env: append(defaultEnvVars, ...getEnvVarFromRequest(request))
```

Reply:
ok